### ISSUE TITLE
no confirmation when a payment is being done

### ISSUE THREAD

**Issue by nicolasburtey** [GaloyMoney/galoy-pay#39]
There is no automatic payment confirmation currently.

Step to reproduce:
- open https://ln.bitcoinbeach.com/nicolas
- pay one sats
- page doesn't refresh

**Comment by leesalminen** [comment]
I just tried this and was unable to reproduce. Enjoy your sats, @nicolasburtey ;).

**Comment by krtk6160** [comment]
> There is no automatic payment confirmation currently.
> 
> Step to reproduce:
> 
> * open https://ln.bitcoinbeach.com/nicolas
> * pay one sats
> * page doesn't refresh

Could you post log output from the browser console? I've been facing this issue as well on my laptop with brave browser, but with firefox it works perfectly. Are you also using brave by any chance?

My log output shows this error -> `WebSocket connection to '<URL>' failed: WebSocket is closed before the connection is established.`

**Comment by nicolasburtey** [comment]
I am using brave yes

**Comment by leesalminen** [comment]
I tried this out using Brave `Version 1.33.106 Chromium: 96.0.4664.110 (Official Build) (arm64)` on MacOS 12.0.1 a couple times and was unable to reproduce. Initially I was testing with Safari 15.1 and Firefox 95.0 on the same machine. 

I did find some potentially related issues reported on other projects:

- https://github.com/brave/brave-browser/issues/15410
- https://github.com/hasura/graphql-engine/issues/4509

On https://github.com/hasura/graphql-engine/issues/4509 someone mentioned that adding `lazy: true` to the `new WebSocketLink()` `options` object fixed the issue for them.

On the whole though, this seems like an issue with Brave Browser with many different developers reporting it. The only truly reliable way to resolve it seems to be with the user closing and re-opening the browser.
